pkg/cleaner/cleaner.go

```diff
@@ -89,36 +90,35 @@ func (c *Cleaner) RemoveCRDs(releaseNamespace, controllerManagerName string, web
 func (c *Cleaner) scaleDownController(namespace, controllerName string) error {
 	klog.V(4).Infof("Fetching deployment %s/%s", namespace, controllerName)
 	deployment, err := c.client.AppsV1().Deployments(namespace).Get(context.Background(), controllerName, v1.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("cannot get deployment %s/%s: %s", namespace, controllerName, err)
-	}
-
-	klog.V(4).Info("Scaling down deployment to zero")
-	replicas := int32(0)
-	deploymentCopy := deployment.DeepCopy()
-	deploymentCopy.Spec.Replicas = &replicas
-	_, err = c.client.AppsV1().Deployments(deploymentCopy.Namespace).Update(context.Background(), deploymentCopy, v1.UpdateOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to update deployment %s/%s: %v", namespace, controllerName, err)
-	}
-
-	err = wait.Poll(3*time.Second, 120*time.Second, func() (done bool, err error) {
-		klog.V(4).Info("Waiting for deployment scales down...")
-		deployment, err := c.client.AppsV1().Deployments(namespace).Get(context.Background(), controllerName, v1.GetOptions{})
+	if err == nil {
```

Comment:
in Go, the happy path should be aligned to the left. Also, we need to adjust the error handling - skip scaling only in case of `not found` and log a proper info message. In case of any other error we need to return it (btw. if err != nil then `deployment` will be nil so in the current implementation it will result in panic). This is an automated migration job so the steps need to be logged just in case of a debug

```go
import apierrors "k8s.io/apimachinery/pkg/api/errors"
// ...

	switch {
	case err == nil:
	case apierrors.IsNotFound(err):
		klog.V(4).Info("Deployment not found. Skippking scale down to zero.")
		return nil
	default:
		return fmt.Errorf("cannot get deployment %s/%s: %s", namespace, controllerName, err)
	}

// rest of the happy path
```

Reply:
Put in your changes, lemme know what you think!

Reply:
nice, lgtm 👍